recipes/ckanweb-deploy-exts.rb

```diff
@@ -202,7 +177,73 @@
 			EOS
 		end
 	end
+end
+
+plugin_url_string = plugin_urls.join(' ')
+
+log "#{DateTime.now}: Installing plugin eggs"
+execute "Install plugins" do
+	user account_name
+	group account_name
+	command "#{pip} install #{plugin_url_string}"
+end
+
+log "#{DateTime.now}: Installing plugin requirement files"
+bash "Install plugin requirements" do
+	cwd "#{virtualenv_dir}/src"
+	code <<-EOS
+		PYTHON_MAJOR_VERSION=$(#{python} -c "import sys; print(sys.version_info.major)")
+		PYTHON_REQUIREMENTS_FILE=requirements-py$PYTHON_MAJOR_VERSION.txt
+		CKAN_MINOR_VERSION=$(#{python} -c "import ckan; print(ckan.__version__)" | grep -o '^[0-9]*[.][0-9]*')
+		CKAN_REQUIREMENTS_FILE=requirements-$CKAN_MINOR_VERSION.txt
+		for extension in `ls -d ckanext-*`; do
+			if [ -f $extension/$PYTHON_REQUIREMENTS_FILE ]; then
+				REQUIREMENTS_FILES="$REQUIREMENTS_FILES -r $PYTHON_REQUIREMENTS_FILE"
+			else
+				if [ -f "$extension/$CKAN_REQUIREMENTS_FILE" ]; then
+					REQUIREMENTS_FILES="$REQUIREMENTS_FILES -r $CKAN_REQUIREMENTS_FILE"
+				else
+					REQUIREMENTS_FILES="$REQUIREMENTS_FILES -r requirements.txt"
+				fi
+			fi
+			# ckanext-harvest uses this filename
+			if [ -f "$extension/pip-requirements.txt" ]; then
+				REQUIREMENTS_FILES="$REQUIREMENTS_FILES -r pip-requirements.txt"
+			fi
+		done
+		#{pip} install $REQUIREMENTS_FILES
```

Comment:
Can you add comments on what this is doing.  From what i can tell, you are flattening all requirements into one pip call. are you also handling cases where a plugin or extension is in the list twice but one is older than the other?

Reply:
If there is an actual incompatibility, then we have bigger problems and we need to fix it. If there is not, then defining the same library multiple times with different version requirements is valid.